crates/sui-types/src/crypto.rs

```diff
@@ -1368,6 +1368,9 @@ where
         // Remove name tag before deserialization using BCS
         let name = serde_name::trace_name::<Self>().expect("Self should be a struct or an enum");
         let name_byte_len = format!("{}::", name).bytes().len();
+        if name_byte_len >= bytes.len() {
+            anyhow::bail!("Failed to deserialize signable bytes {name}");
```

Comment:
For logging purposes I'd also add that `shorter that expected payload was received` and we might attach the length received (`bytes.len()`).